aten/src/ATen/cpu/vec/vec512/vec512_base.h

```diff
@@ -0,0 +1,910 @@
+          typename std::enable_if<c10::is_complex<T>::value, int>::type = 0>
+Vectorized<T> inline maximum(const Vectorized<T> &a, const Vectorized<T> &b) {
+  Vectorized<T> c = Vectorized<T>();
+  for (int i = 0; i != Vectorized<T>::size(); i++) {
+    c[i] = (std::abs(a[i]) > std::abs(b[i])) ? a[i] : b[i];
+    if (_isnan(a[i])) {
+      // If either input is NaN, propagate a NaN.
+      // NOTE: The case where b[i] was NaN is handled correctly by the naive
+      // ternary operator above.
+      c[i] = a[i];
+    }
+  }
+  return c;
+}
+
+// Implements the IEEE 754 201X `minimum` operation, which propagates NaN if
+// either input is a NaN.
+template <class T,
+          typename std::enable_if<!c10::is_complex<T>::value, int>::type = 0>
+Vectorized<T> inline minimum(const Vectorized<T> &a, const Vectorized<T> &b) {
+  Vectorized<T> c = Vectorized<T>();
+  for (int i = 0; i != Vectorized<T>::size(); i++) {
+    c[i] = (a[i] < b[i]) ? a[i] : b[i];
+    if (_isnan(a[i])) {
+      // If either input is NaN, propagate a NaN.
+      // NOTE: The case where b[i] was NaN is handled correctly by the naive
+      // ternary operator above.
+      c[i] = a[i];
+    }
+  }
+  return c;
+}
+
+template <class T,
+          typename std::enable_if<c10::is_complex<T>::value, int>::type = 0>
+Vectorized<T> inline minimum(const Vectorized<T> &a, const Vectorized<T> &b) {
+  Vectorized<T> c = Vectorized<T>();
+  for (int i = 0; i != Vectorized<T>::size(); i++) {
+    c[i] = (std::abs(a[i]) < std::abs(b[i])) ? a[i] : b[i];
+    if (_isnan(a[i])) {
+      // If either input is NaN, propagate a NaN.
+      // NOTE: The case where b[i] was NaN is handled correctly by the naive
+      // ternary operator above.
+      c[i] = a[i];
+    }
+  }
+  return c;
+}
+
+template <class T,
+          typename std::enable_if<!c10::is_complex<T>::value, int>::type = 0>
+Vectorized<T> inline clamp(const Vectorized<T> &a, const Vectorized<T> &min_Vectorized, const Vectorized<T> &max_Vectorized) {
+  Vectorized<T> c = Vectorized<T>();
+  for (int i = 0; i != Vectorized<T>::size(); i++) {
+    c[i] = std::min(std::max(a[i], min_Vectorized[i]), max_Vectorized[i]);
+  }
+  return c;
+}
+
+template <class T,
+          typename std::enable_if<!c10::is_complex<T>::value, int>::type = 0>
+Vectorized<T> inline clamp_max(const Vectorized<T> &a, const Vectorized<T> &max_Vectorized) {
+  Vectorized<T> c = Vectorized<T>();
+  for (int i = 0; i != Vectorized<T>::size(); i++) {
+    c[i] = a[i] > max_Vectorized[i] ? max_Vectorized[i] : a[i];
+  }
+  return c;
+}
+
+template <class T,
+          typename std::enable_if<!c10::is_complex<T>::value, int>::type = 0>
+Vectorized<T> inline clamp_min(const Vectorized<T> &a, const Vectorized<T> &min_Vectorized) {
+  Vectorized<T> c = Vectorized<T>();
+  for (int i = 0; i != Vectorized<T>::size(); i++) {
+    c[i] = a[i] < min_Vectorized[i] ? min_Vectorized[i] : a[i];
+  }
+  return c;
+}
+
+struct Vectorizedi;
+
+#ifdef CPU_CAPABILITY_AVX2

```

Comment:
This must be CPU_CAPABILITY_AVX512, mustn't it?
```suggestion
#ifdef CPU_CAPABILITY_AVX512

```

Reply:
Thanks! That's a blunder!!!